File: online-judge-solutions/Codeforces/1462B.cpp
//https://codeforces.com/problemset/problem/1462/B
//Lang		:	GNU C++14
//Time		:	93 ms
//Memory	:	0 KB
#include<iostream>
using namespace std;
int main(){
	int t,n;
	cin>>t;
	while(t--){
		string s,o="2020";
		cin>>n>>s;
		if(s==o) cout<<"YES\n";
		else{
			int t=1;
			for(int i=0;i<n && t;i++){
				string r=s;
				if(o==r.erase(i,n-4)) t=0;
			}
			t==0? cout<<"YES\n":cout<<"NO\n";
		}
	}
	return 0;
}

File: online-judge-solutions/Codeforces/1462B.py
#https://codeforces.com/problemset/problem/1462/B
#Lang		:	Python 3.8
#Time		:	155 ms
#Memory     :	0 KB
t=int(input())
for x in range(0,t):
    n,s,o,t=int(input()),input(),"2020",1
    if(s==o):print("YES")
    else:
        for i in range(n):
            r=s
            r=r[0:i]+r[i+n-4:]
            if(o==r): t=0
        print("YES") if t==0 else print("NO")
